### ISSUE TITLE
Using eslintIgnore to only check js file doesn't work with subfolders

### ISSUE THREAD

**Issue by unional** [eslint/eslint#10274]
**Tell us about your environment**

* **ESLint Version:**  4.18.2
* **Node Version:** 9.11.1
* **npm Version:** 5.6.0

.eslintignore
```sh
**/*
!**/*.js
```

```js
const Eslint = require('eslint').CLIEngine;
const eslint = new Eslint();
console.log(eslint.isPathIgnored('index.js')) // false, correct
console.log(eslint.isPathIgnored('x/index.js')) // true, incorrect
```

Also tried:
```sh
*
!*.js
```
same result.

**Comment by platinumazure** [comment]
Hi @unional, thanks for the issue.

You shouldn't need to use `.eslintignore` to restrict by extension.

Instead, you can use the following CLI options:

* `--ext`: Specify what file extensions should be searched for when directory names are passed in.
* Globs: Pass in a glob as an argument to ESLint to restrict your search (e.g., `**/*.js`)

Or the following CLIEngine options:

* `extensions` (on constructor): Specify what file extensions should be searched for when directory names are passed in.
* Globs in `executeOnFiles`: Pass in glob patterns such as `**/*.js` to only execute on files matching the glob pattern.

-------------

If you've already tried the above, could you please let us know what output/results you saw and what you were expecting to see?

**Comment by unional** [comment]
Thanks.
So seems like this is no longer valid: https://github.com/eslint/eslint-cli/issues/1

**Comment by nzakas** [comment]
It looks like the conversation is stalled here. As this is a question rather than an action item, I'm closing the issue. If you still need help, please send a message to our [mailing list](https://groups.google.com/group/eslint) or [chatroom](https://gitter.im/eslint/eslint). Thanks!
